server/internal/usecase/interfaces/asset.go

```diff
@@ -61,6 +61,7 @@ type Asset interface {
 	FindByIDs(context.Context, []id.AssetID, *usecase.Operator) (asset.List, error)
 	FindByProject(context.Context, id.ProjectID, AssetFilter, *usecase.Operator) (asset.List, *usecasex.PageInfo, error)
 	FindFileByID(context.Context, id.AssetID, *usecase.Operator) (*asset.File, error)
+	FindFilesByIDs(context.Context, id.AssetIDList, *usecase.Operator) (map[id.AssetID]*asset.File, error)
```

Comment:
```suggestion
	FindFilesByIDs(context.Context, id.AssetIDList, *usecase.Operator) (asset.FileList, error)
```
return list, and implement a method in the list to return a map

Reply:
Since `asset.File` does not have assetId it will be impossible to tell which file is for which asset when returning `asset.FileList` that is why I am returning a map with asset id as a key.

```
type File struct {
	name        string
	size        uint64
	contentType string
	path        string
	children    []*File
	files       []*File
}
```

Reply:
the asset file document (which is stored in DB) contains the asset ID, we can expose it to the model 🤔

Reply:
Can you explain why is it not a good idea to use a map?